iota-execution/latest/iota-adapter/src/execution_engine.rs

```diff
@@ -783,30 +784,39 @@ mod checked {
     pub fn construct_advance_epoch_pt(
         mut builder: ProgrammableTransactionBuilder,
         params: &AdvanceEpochParams,
+        protocol_version: u64,
     ) -> Result<ProgrammableTransaction, ExecutionError> {
-        // Step 1: Create storage charges and computation rewards.
-        let (storage_charges, computation_rewards) = mint_epoch_rewards_in_pt(&mut builder, params);
+        // Step 1: Create storage and computation charges.
+        let (storage_charges, computation_charges) = mint_epoch_rewards_in_pt(&mut builder, params);
 
         // Step 2: Advance the epoch.
         let mut arguments = vec![
             builder
-                .pure(params.validator_target_reward)
+                .pure(params.validator_subsidy)
                 .expect("bcs encoding a u64 should not fail"),
             storage_charges,
-            computation_rewards,
+            computation_charges,
         ];
-        let call_arg_arguments = vec![
+
+        let mut call_arg_vec = vec![
+            CallArg::Pure(bcs::to_bytes(&params.computation_charge_burned).unwrap()),
             CallArg::IOTA_SYSTEM_MUT,
             CallArg::Pure(bcs::to_bytes(&params.epoch).unwrap()),
             CallArg::Pure(bcs::to_bytes(&params.next_protocol_version.as_u64()).unwrap()),
             CallArg::Pure(bcs::to_bytes(&params.storage_rebate).unwrap()),
             CallArg::Pure(bcs::to_bytes(&params.non_refundable_storage_fee).unwrap()),
             CallArg::Pure(bcs::to_bytes(&params.reward_slashing_rate).unwrap()),
             CallArg::Pure(bcs::to_bytes(&params.epoch_start_timestamp_ms).unwrap()),
-        ]
-        .into_iter()
-        .map(|a| builder.input(a))
-        .collect::<Result<_, _>>();
+        ];
+        if protocol_version > 1 {
+            call_arg_vec.push(CallArg::Pure(
+                bcs::to_bytes(&params.computation_charge_burned).unwrap(),
+            ));
+        }
+        let call_arg_arguments = call_arg_vec
```

Comment:
Is this ok as a way to decide whether to include the computation charges burned as an argument? Alternative is to use the "protocol_defined_base_fee" flag from the protocol config. We discussed this but I do not recall the conclusion, if there was one.